server/chess.py

```diff
@@ -85,14 +96,10 @@ def get_moves_in_direction(
         while True:
             file += file_direction
             rank += rank_direction
-            target = self.get_piece(rank, file)
-            if target:
-                if target.side == piece.side:
-                    break
+            if validate_move_piece(self, piece, rank, file):
                 yield rank, file
-                break
             else:
-                yield rank, file
+                break
```

Comment:
This does not cover the case where a move would be invalid but travelling further would be valid (eg. travelling to protect the king)